File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.js
function swap(a, b) {
    return [b, a]
}


function solution(n,a,b)
{
    if (a > b) {
        [a, b] = swap(a, b)
    }
    let curr_round = 1;
    
    while (1) {
        // 현재 라운드에서 만나는 경우
        if (a % 2 === 1 && a + 1 === b) break
        
        // 다음 부여받을 번호 결정
        if (a % 2 === 0) {
            a /= 2
        } else {
            a = parseInt(a / 2) + 1
        }
        
        if (b % 2 === 0) {
            b /= 2
        } else {
            b = parseInt(b / 2) + 1
        }
        
        curr_round++
    }

    return curr_round;
}

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.py
def swap(a, b):
    return b, a


def solution(n,a,b):
    if a > b:
        a, b = swap(a, b)
        
    curr_round = 1
    while True:
        # 현재 라운드에 만나는 경우
        if a % 2 == 1 and a + 1 == b:
            break
                
        # 다음 부여받을 번호 결정
        a = a // 2 if a % 2 == 0 else (a // 2) + 1
        b = b // 2 if b % 2 == 0 else (b // 2) + 1
            
        curr_round += 1
        
    return curr_round
                
        
                